crates/polars-io/src/file_cache/eviction.rs

```diff
@@ -64,17 +196,40 @@ impl EvictionManager {
                     },
                 }
 
-                tokio::time::sleep(Duration::from_secs(sleep_interval)).await;
+                loop {
+                    let min_ttl = self.min_ttl.load(std::sync::atomic::Ordering::Relaxed);
+                    let sleep_interval = std::cmp::max(min_ttl / 4, {
+                        #[cfg(debug_assertions)]
+                        {
+                            3
+                        }
+                        #[cfg(not(debug_assertions))]
+                        {
+                            60
+                        }
+                    });
+
+                    let since_last_eviction =
+                        Instant::now().duration_since(last_eviction_time).as_secs();
+                    let sleep_interval = sleep_interval.saturating_sub(since_last_eviction);
+                    let sleep_interval = Duration::from_secs(sleep_interval);
+
+                    tokio::select! {
+                        _ = self.notify_ttl_updated.notified() => {
+                            continue;
+                        }
+                        _ = tokio::time::sleep(sleep_interval) => {
+                            break;
+                        }
+                    }
+                }
             }
         });
     }
 
     fn update_file_list(&mut self) -> PolarsResult<()> {
-        let data_dir = &self.prefix.join("d/");
-        let metadata_dir = &self.prefix.join("m/");
-
-        let data_files_iter = std::fs::read_dir(data_dir).unwrap();
-        let metadata_files_iter = std::fs::read_dir(metadata_dir).unwrap();
+        let data_files_iter = std::fs::read_dir(self.data_dir.as_ref()).unwrap();
```

Comment:
Can we error here? This can be cause by a non-implementation error.

Reply:
I think I can keep it as unwrap, as I maintain an invariant that those directories are created during initialization - I have hoisted their creation to the same place that the `EvictionManager` is created to make it clearer

Reply:
Yes, but someone could remove the directory under our nose right?

Reply:
ah.. 🥲